### PR TITLE
feat(projects): Only show project name in project selector (internal only)

### PR DESCRIPTION
**Pull request by macqueen**

### CODE REVIEW

File: src/sentry/static/sentry/app/components/projectHeader/projectSelector.jsx [hunk 1]
```diff
@@ -220,7 +222,13 @@ const ProjectSelector = createReactClass({
 
   getProjectLabel(team, project, hasSingleTeam, highlightText) {
     let label, text;
-    if (!hasSingleTeam && project.name.indexOf(team.name) === -1) {
+    let features = new Set(this.props.organization.features);
+
+    if (
+      !hasSingleTeam &&
+      project.name.indexOf(team.name) === -1 &&
+      !features.has('internal-catchall')
```
Comment (MaxBittker) [inline src/sentry/static/sentry/app/components/projectHeader/projectSelector.jsx]: One question - in the new, internal behavior, do we want to show `slug` or `name`?

Comment (macqueen) [reply src/sentry/static/sentry/app/components/projectHeader/projectSelector.jsx]: mm i remember that conversation happening and do not remember what the decision was :-/

File: src/sentry/static/sentry/app/components/projectHeader/projectSelector.jsx [hunk 2]
```diff
@@ -309,12 +317,20 @@ const ProjectSelector = createReactClass({
...
     let org = this.props.organization;
+    let features = new Set(org.features);
     let access = new Set(org.access);
     let hasSingleTeam = org.teams.length === 1;
 
-    let projectList = sortArray(this.state.projectList, ([team, project]) => {
-      return [!project.isBookmarked, team.name, project.name];
-    });
+    let projectList;
+    if (features.has('internal-catchall')) {
```
Comment (MaxBittker) [inline src/sentry/static/sentry/app/components/projectHeader/projectSelector.jsx]: i would say something abt duplicating the logic in a slightly nicer way, but i guess this temporary code anyway

**Review by MaxBittker — APPROVED** [review]
looks good to me 👍

File: src/sentry/static/sentry/app/components/projectHeader/projectSelector.jsx (hunk 1 above)
Comment (macqueen) [reply src/sentry/static/sentry/app/components/projectHeader/projectSelector.jsx]: ok we're keeping name

File: src/sentry/static/sentry/app/components/projectHeader/projectSelector.jsx (hunk 2 above)
Comment (macqueen) [reply src/sentry/static/sentry/app/components/projectHeader/projectSelector.jsx]: yeah the pr i merged yesterday also has a decent amount of this grossness. i couldn't come up with anything that clever and figure it should only be around for a month or so, but if you have suggestions, i'm down!

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

**Pull request merged**